### ISSUE TITLE
AO and SSR are still different from VPX

### ISSUE THREAD

**Issue by toxieainc** [vpinball/vpvr#40]
I know that you're still busy with all the conversions and fixes, but just to mention/tracking it..
SSR is showing some black artifacts and AO is weirdly flickering. SSR i diffed with VPX, but couldn't find an obvious cause so far.  :/
Test table was JP Salas Star Trek (Stern).

**Comment by vbousquet** [comment]
The texture sampling is not yet cleaned up. I expect it to solve this when it will be fully clean. I will commit first part this evening. (The rest will have to wait a bit since I will be away for my work the next few days).

**Comment by vbousquet** [comment]
Pushed in PR https://github.com/vpinball/vpvr/pull/41

It's not yet finished. Now that we have a clean sampler state management, I need to review the different place wher VPX change this state (not fully clean on that side too since I think there is a mix between D3DXEffect framework state management, and direct state management which I think is somewhat error prone and difficult to read) to align the codebases. Then I will add anisotropic filtering (easy one). Then I will perform some testing to be sure rendering matches

**Comment by toxieainc** [comment]
Yes, the DX state stuff is a complete mess.  :/
But it evolved from DX7 in that whacky fashion (as DX7 was even more of a mess in other ways), so sorry for that.  ;)
Thanks for being so brave to tackle all of that!!

(sidenote: with PR 41 both AO and SSR do not seem to do anything at the moment for me)

**Comment by vbousquet** [comment]
You're right, I broke it again... Will have a look. This sampler setup must be really clean and maintainable for a better future...

**Comment by toxieainc** [comment]
No hurry. I will be away and only loosely checkin in on VPX/VR the next weeks anyway.
